src/commands/find.spec.ts

```diff
@@ -0,0 +1,133 @@
+    MockPreferenceStorage
+  )
+  container.registerSingleton<ITranslationFiles>(
+    'TranslationFiles',
+    TranslationFiles
+  )
+
+  fileSystem = container.resolve('FileSystem')
+  codeParser = container.resolve('CodeParser')
+})
+
+const whenKeyIsFoundInCodeAndTranslationFiles = prepare
+  .do(() => {
+    sinon.stub(fileSystem, 'readJSON').resolves({
+      Page_title: 'Título da Página',
+    })
+
+    sinon
+      .stub(codeParser, 'getKeyOccurrence')
+      .resolves(
+        new KeyOccurrence('Page_title', 'app/message-pin/client/pinMessage.js')
+      )
+  })
+  .stdout()
+  .command(['find', 'Page_title', '--lang', 'pt-BR'])
+
+const whenKeyIsFoundInCodeButNotInTranslationFiles = prepare
+  .do(() => {
+    sinon.stub(fileSystem, 'readJSON').resolves({})
+    sinon
+      .stub(codeParser, 'getKeyOccurrence')
+      .resolves(
+        new KeyOccurrence('away_female', 'app/message-pin/client/pinMessage.js')
+      )
+  })
+  .stdout()
+  .command(['find', 'away_female', '--lang', 'pt-BR'])
+
+const whenKeyIsFoundInTranslationFilesButNotInCode = prepare
+  .do(() => {
+    sinon
+      .stub(fileSystem, 'readJSON')
+      .resolves({ to_infinity_and_beyond: 'Ao infinito, e além!' })
+
+    sinon.stub(codeParser, 'getKeyOccurrence').resolves(undefined)
+  })
+  .stdout()
+  .command(['find', 'to_infinity_and_beyond', '--lang', 'pt-BR'])
+
+const whenKeyIsNotFoundAnywhere = prepare
+  .do(() => {
+    sinon.stub(fileSystem, 'readJSON').resolves({})
+    sinon.stub(codeParser, 'getKeyOccurrence').resolves(undefined)
+  })
+  .stdout()
+  .command(['find', 'away_male', '--lang', 'pt-BR'])
+
+describe('find command', () => {
+  whenKeyIsFoundInCodeAndTranslationFiles.it(
+    'should say the language is ok',
+    (ctx) => {
+      expect(ctx.stdout).to.contain('Page_title is OK')
+    }
+  )
+
+  whenKeyIsFoundInCodeAndTranslationFiles.it(
+    'should say where the key was found',
+    (ctx) => {
+      expect(ctx.stdout).to.contain(
+        'It is referenced in the codebase (found in app/message-pin/client/pinMessage.js)'
+      )
+    }
+  )
+
+  whenKeyIsFoundInCodeAndTranslationFiles.it(
+    'should return the translation found',
+    (ctx) => {
+      expect(ctx.stdout).to.contain(
+        "It has a translation in pt-BR: 'Título da Página'"
+      )
+    }
+  )
+
+  whenKeyIsFoundInCodeButNotInTranslationFiles.it(
+    'should say the key is untranslated',
+    (ctx) => {
+      expect(ctx.stdout).to.contain('away_female is UNTRANSLATED')
+    }
+  )
+
+  whenKeyIsFoundInTranslationFilesButNotInCode.it(
+    'should say the key is unused',
+    (ctx) => {
+      expect(ctx.stdout).to.contain('to_infinity_and_beyond is UNUSED')
+      expect(ctx.stdout).to.contain(
+        'It has a translation in pt-BR, but it is not used in the codebase.'
+      )
+    }
+  )
+
+  whenKeyIsNotFoundAnywhere.it('should say the key was not found', (ctx) => {
+    expect(ctx.stdout).to.contain('away_male was not found')
+  })
+})
```

Comment:
I'm starting to think that this "fancy test" approach is not even close to fancy. Lots of boilerplate to write some tests. Maybe I'll stick with good old describes...